TypeScript.py

```diff
@@ -439,6 +445,49 @@ def checkUpdateView(view):
         if cli.reloadRequired(view):
             reloadBuffer(view, clientInfo)
 
+# these three methods work with signature help
+def handle_signature_help(signature_help_response):
+    if signature_help_response.success:
+        view = active_view()
+        if is_typescript(view):
+            clientInfo = cli.getOrAddFile(view.file_name())
+            applicableRegion = span_to_region(signature_help_response.body.applicableSpan, view, 1)
+            view.add_regions("signatureApplicable", [applicableRegion], flags=sublime.HIDDEN)
+            print("signature applicable to " + view.substr(applicableRegion))
+            if clientInfo.method_tip_active:
+                # update method tip UI
+                print("update method tip UI")
+            else:
+                # put up method tip UI
+                clientInfo.method_tip_active = True
+                print("show method tip UI")
+
+# check trigger for signature help
+def check_trigger_key(key, view):
+    if cli.ST2():
+        return
+    print('check trigger key ' + key)
+    if (key == '(') or (key == ','):
```

Comment:
For the record in VS we also have trigger characters to support tagged templates. I can look into it tomorrow.